src/objects.c

```diff
@@ -1877,6 +1880,143 @@ Obj FuncFORCE_SWITCH_OBJ(Obj self, Obj obj1, Obj obj2) {
+*/
+
+Obj FuncTNUM_OBJ (
+    Obj                 self,
+    Obj                 obj )
+{
+    Obj                 res;
+    Obj                 str;
+    const Char *        cst;
+
+    res = NEW_PLIST( T_PLIST, 2 );
+    SET_LEN_PLIST( res, 2 );
+
+    /* set the type                                                        */
+    SET_ELM_PLIST( res, 1, INTOBJ_INT( TNUM_OBJ(obj) ) );
+    cst = TNAM_OBJ(obj);
+    str = MakeImmString(cst);
+    SET_ELM_PLIST( res, 2, str );
+
+    /* and return                                                          */
+    return res;
+}
+
+Obj FuncTNUM_OBJ_INT (
+    Obj                 self,
+    Obj                 obj )
+{
+
+  
+    return  INTOBJ_INT( TNUM_OBJ(obj) ) ;
+}
+
+/****************************************************************************
+**
+*F  FuncOBJ_HANDLE( <self>, <obj> ) . . . . . .  expert function 'OBJ_HANDLE'
+*/
+Obj FuncOBJ_HANDLE (
+    Obj                 self,
+    Obj                 obj )
+{
+    UInt                hand;
+    UInt                prod;
+    Obj                 rem;
+
+    if ( IS_INTOBJ(obj) ) {
+        return (Obj)INT_INTOBJ(obj);
+    }
+    else if ( TNUM_OBJ(obj) == T_INTPOS ) {
+        hand = 0;
+        prod = 1;
+        while ( EQ( obj, INTOBJ_INT(0) ) == 0 ) {
+            rem  = RemInt( obj, INTOBJ_INT( 1 << 16 ) );
+            obj  = QuoInt( obj, INTOBJ_INT( 1 << 16 ) );
+            hand = hand + prod * INT_INTOBJ(rem);
+            prod = prod * ( 1 << 16 );
+        }
+        return (Obj) hand;
+    }
+    else {
+        ErrorQuit( "<handle> must be a positive integer", 0L, 0L );
+        return (Obj) 0;
+    }
+}
+
+
+/****************************************************************************
+**
+*F  FuncHANDLE_OBJ( <self>, <obj> ) . . . . . .  expert function 'HANDLE_OBJ'
+**
+**  This is a very quick function which returns a unique integer for each object
+**  non-identical objects will have different handles. The integers may be large.
+*/
+Obj FuncHANDLE_OBJ (
+    Obj                 self,
+    Obj                 obj )
+{
+    Obj                 hnum;
+    Obj                 prod;
+    Obj                 tmp;
+    UInt                hand;
+
+    hand = (UInt) obj;
+    hnum = INTOBJ_INT(0);
+    prod = INTOBJ_INT(1);
+    while ( 0 < hand ) {
+        tmp  = PROD( prod, INTOBJ_INT( hand & 0xffff ) );
+        prod = PROD( prod, INTOBJ_INT( 1 << 16 ) );
+        hnum = SUM(  hnum, tmp );
+        hand = hand >> 16;
+    }
+    return hnum;
+}
+
+/* This function does quite  a similar job to HANDLE_OBJ, but (a) returns 0
+   for all immediate objects (small integers or ffes) and (b) returns reasonably
+   small results (roughly in the range from 1 to the max number of objects that
+   have existed in this session. In HPCGAP it just returns the same value as
+   HANDLE_OBJ for non-immediate objects */
+
+Obj FuncMASTER_POINTER_NUMBER(Obj self, Obj o)
```

Comment:
I wonder if these are really functions about objects? Yes, most of them have an "OBJ" in their name. But in reality, they all seem to be about *bags* not objects.  At the very least, this is true for `FuncMASTER_POINTER_NUMBER`, which is about an implementation details of GASMAN... so that would speak for moving the code to `gasman.c` -- except that there, one cannot easily install function handlers....

Anyway, this is kind of bikeshedding (though I'd argue that moving the functions in the first place is so, too). I don't mind moving the functions here, I just wanted to share my thoughts.